src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionSymbolService.cs

```diff
@@ -12,23 +13,36 @@ internal abstract class AbstractGoToDefinitionSymbolService : IGoToDefinitionSym
     {
         protected abstract ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation);
 
-        public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, CancellationToken cancellationToken)
+        public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, bool includeType, CancellationToken cancellationToken)
         {
             var workspace = document.Project.Solution.Workspace;
 
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var semanticInfo = await SymbolFinder.GetSemanticInfoAtPositionAsync(semanticModel, position, workspace, cancellationToken).ConfigureAwait(false);
+            var symbol = GetSymbol(semanticInfo, includeType);
 
-            // prefer references to declarations.  It's more likely that the user is attempting to 
-            // go to a definition at some other location, rather than the definition they're on.  
+            if (symbol is null)
+            {
+                return (null, semanticInfo.Span);
+            }
+
+            return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
+        }
+
+        private ISymbol GetSymbol(TokenSemanticInfo semanticInfo, bool includeType)
+        {
```

Comment:
could be reduced to (if you want)
```c#
=> semanticInfo.AliasSymbol
?? semanticInfo.ReferencedSymbols.FirstOrDefault()
?? semanticInfo.DeclaredSymbol
?? (includeType ? semanticInfo.Type : null);
```

Reply:
forget about it. i forgot about the comment

Reply:
yep, it makes more sense. I did it this way so I was able to put a breakpoint there while debugging